### PR TITLE
feature: Add re:Invent 2020 features

### PR DESCRIPTION
**Pull request by ChoiByungWook**
*Issue #, if available:*

*Description of changes:*
* feature: SDK support for MXNet 1.8.0
* feature: SDK support for SageMaker Pipelines
* feature: SDK support for SageMaker Lineage
* feature: SDK support for SageMaker Feature Store
* feature: SDK support for SageMaker Model Registry
* feature: SDK support for SageMaker Projects

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.16.4.dev0
+2.17.0.dev0
```
Comment (ajaykarpur) [inline VERSION:1]: This file should not be changed manually.
```suggestion
2.16.4.dev0
```

**Review by ajaykarpur — CHANGES_REQUESTED** [review]

**Review by ajaykarpur — APPROVED** [review]

File: VERSION (hunk 1 above)
Comment (ajaykarpur) [reply VERSION:1]: Discussed offline.

Comment (ChoiByungWook) [reply VERSION:1]: Agreed. Following the pattern used last year: https://github.com/aws/sagemaker-python-sdk/pull/1143/commits/dcd5d437bf2338ddba5c49cf154fda83482a1264

1.45.0 was still published for PyPI: https://pypi.org/project/sagemaker/1.45.0/

**Review by metrizable — APPROVED** [review]

File: src/sagemaker/model_metrics.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        model_statistics=None,
+        model_constraints=None,
+        model_data_statistics=None,
+        model_data_constraints=None,
+        bias=None,
+        explainability=None,
+    ):
+        """Initialize a ``ModelMetrics`` instance and turn parameters into dict.
+
+        # TODO: flesh out docstrings
```
[the remark this answers is not in the document]
Comment (ChoiByungWook) [reply src/sagemaker/model_metrics.py:31]: These can be addressed after this PR.

**Review by icywang86rui — APPROVED** [review]

**Comment by ChoiByungWook** [comment]
> ### AWS CodeBuild CI Report
> * CodeBuild project: sagemaker-python-sdk-pr
> * Commit ID: [96ec1f3](https://github.com/aws/sagemaker-python-sdk/commit/96ec1f3d546535fd72b2106e33bb58d5ce5d92d9)
> * Result: FAILED
> * [Build Logs](https://wsr3865w1k.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c0e3989c-108f-4279-b083-c649988493d5%2Fbuild.log) (available for 30 days)
> 
> _Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)_

[test_integ_history_server](https://github.com/aws/sagemaker-python-sdk/blob/9324af9fa150cce9b0a42f2dee5f94707f9fae64/tests/integ/test_spark_processing.py#L292) failed due to
```
>           assert response.status == 200
E           assert 502 == 200
E            +  where 502 = <urllib3.response.HTTPResponse object at 0x7f1f5bc51d00>.status
```

This test didn't fail before.

**Comment by ChoiByungWook** [comment]
export IGNORE_COVERAGE=- ; tox -e py36 -- -s -vv tests/integ/test_spark_processing.py::test_integ_history_server ; unset IGNORE_COVERAGE
_________________________________________________________________________ summary _________________________________________________________________________
  py36: commands succeeded
  congratulations :)

Overriding test check, as the failure was transient.

**Pull request merged**